### ISSUE TITLE
Allow running Effekt programs directly from VS Code

### ISSUE THREAD

**Issue by marzipankaiser** [effekt-lang/effekt-vscode#12]
AFAICT, there is currently no way to run Effekt programs from the VS Code GUI. (Apart from opening the terminal and running it directly).

**Comment by b-studios** [comment]
Yes, that's right. Of course, you could use the terminal built into vscode; probably that's what you meant.

Some languages (like Flix) have vscode support where `run` shows up as a codelens over a toplevel function.
For this, we would need some support from our compiler, I'd assume. At least, we could generate a `main` that calls this particular function and run it immediately (like it is done in the Repl).

https://github.com/effekt-lang/effekt/blob/master/effekt/jvm/src/main/scala/effekt/Repl.scala#L293-L321

**Comment by jiribenes** [comment]
This would be already very helpful even if it worked only for `def main(): Unit / {}` and ran the whole file with the currently chosen backend.

**Comment by b-studios** [comment]
For all other functions that take no argument and don't have unhandled effects, it is very easy to generate main as a thin wrapper and can it immediately. In the Repl we do something very similar

**Comment by jiribenes** [comment]
#28 has a crude version that greps for `def main()` in a file and shows a `|> Run` code lens above it that runs the whole file like `effekt $CURRENT_FILE --backend $BACKEND`.
There was also an attempt to open the REPL, but it fails on a few details mentioned in #28. Here's the relevant commit that reverts from REPL back to running a file (you can just apply it in reverse): https://github.com/effekt-lang/effekt-vscode/pull/28/commits/5a12148526e6896f6b27b15af8f30ecd902e1ad2

That's why I don't consider this issue resolved (yet), please feel free to pick this up and continue with a better solution. :)

**Comment by jiribenes** [comment]
Possible heuristic as an improvement: if there are two `def main()`s, do _not_ show the run button (can we show some error there?)
